ccp/index_reader.py

```diff
@@ -126,9 +136,27 @@ def load_project_entry(self, entry):
 
     def get_pipeline_name(self):
         """
-        Returns the pipeline name based on the project object values
+        Returns the pipeline name based on appid, jobid and desired_tag
+        and also converts it to lower case
         """
-        return "{}-{}-{}".format(self.app_id, self.job_id, self.desired_tag)
+        pipeline_name = "{}-{}-{}".format(
+            self.app_id, self.job_id, self.desired_tag).lower()
+
+        # pipeline name which becomes value for metadata.name field in template
+        # must confront to following regex as per oc
+        # We tried to make the string acceptable by converting it to lower case
+        # Below we are adding another gate to make sure the pipeline_name is as
+        # per requirement, otherwise raising an exception with proper message
+        # to indicate the issue
+        pipeline_name_regex = ("^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]"
+                               "([-a-z0-9]*[a-z0-9])?)*$")
+        match = re.match(pipeline_name_regex, pipeline_name)
+
+        if not match:
+            msg = ("The pipeline name populated {} can't be used in OpenShift "
+                   "template in metadata.name field. ".format(pipeline_name))
+            raise(InvalidPipelineName(msg))
```

Comment:
From the definition of `InvalidPipelineName`, it seems to me that if it's going to encounter a project where this `raise` statement indeed gets executed, it will cause the seed-job to fail with an exception. Am I seeing it correctly? I think we would prefer having a traceback and continue execution of seed-job so that other builds in the batch don't get affected.

Reply:
Adding the `InvalidPipelineName` exception is a part of taking care of buildconfig name. So if that exception arises, shouldn't that be handled in the same PR? I believe it should be handled in this PR only.

Reply:
where are we catching the exception? if we dont have a catch it can potentially restart jenkins. yeah I would also vote for adding the exception handling in the same PR.